### PR TITLE
Fix html/scss editor

### PR DESCRIPTION
**Pull request by qsm-odoo**

### CODE REVIEW

File: addons/web_editor/static/src/scss/web_editor.common.scss [hunk 1]
```diff
@@ -363,12 +373,28 @@ a.o_underline {
             height: 100%;
             cursor: ew-resize;
         }
+    }
+}
 
-        .ace_scroller.o_error::after {
```
Comment (rdeodoo) [inline addons/web_editor/static/src/scss/web_editor.common.scss:367]: Maybe you could keep this one (with same color and same position) so when scrolling down, the visual indication of an error is not lost.
Not easy to explain so here is a step to reproduce:
- Open a file in HTML editor which has scroll
- Remove a closing tag or whatever to produce an error on top of the tile
- Scroll down
Before: the red line was kept
Now: It is gone, only appearing back when the scroll go back on the error

If not clear enough, screenshots :D (error is line 10 or smth)

Before:

image

After:
image

Comment (qsm-odoo) [reply addons/web_editor/static/src/scss/web_editor.common.scss:367]: Good catch, solved as discussed on Discord (showing the error icon at first or last line if the user has to scroll)

**Comment by rdeodoo** [comment]
@robodoo r+

Nice job!

@mart-e May we have a security approval here? There is 2 innerHTML here:
One is just a `\n` to `<br/>` replacement.
The second is just inserting part of the browser native XML parser result, and it is just a move on top of that.

Additional info about this PR: this comes from the fact the HTML Editor is barely usable in real user cases. It either:
1. Won't show there is an error and clicking on save will seems to do nothing
Eg: Having an error of not autoclosing a tag on top of file, save, error is shown but the editor directly scroll to the bottom, making the error disappear (and the vertical red line). So you don't have time to actually see it.
Indeed, on scroll, errors are removed because the editor is re-rendering itself completely.
2. In the best case, it will highlight the line number in red to indicate there is an error, but without any message so no one can figure why. It actually took me 5 years and somehow to show me that you could click on it to display the error inside a modal :grimacing: . No one will figure.
3. Most of the time, the error will be shown at the very bottom of the file, not on the line that created the error, so you don't have a hint of the error, and the editor is misleading you by highlighting a wrong line.

Eg, simply type the following code, which has a non auto-closing `` tag and a `<` symbol.
```html
<img src="whatever.jpg">
<script>
if (0 < 1) {
    alert();
}
</script>
```
As this editor is actually an XML parser, those are not valid, img should absolutely be autoclosed, and `<` should be `&lt;`.
And one won't ever be able to save it despite it being a very common use case: copy / paste a given code snippet that you got from an external service like a banner / a countdown / a google script / etc.

The goal was then to put the HTML Editor where it belongs, in debug mode, as it is by definition an (very) advanced feature, which will just makes people break their DB most of the time anyway.
Since FP doesn't want to do that, this is a best effort to at least make it usable:
1. Error is now preserved when scrolling
2. Error is now shown automatically inside a popover

**Comment by rdeodoo** [comment]
@robodoo rebase-ff

**Comment by mart-e** [comment]
@robodoo override=ci/security
Reason: only replacing `\n` to `<br>`, not evaluating new elements
cc @odoo/rd-security

**Pull request closed without merging**